plugin/pkg/auth/authenticator/token/bootstrap/bootstrap_test.go

```diff
@@ -247,3 +288,75 @@ func TestTokenAuthenticator(t *testing.T) {
 		}()
 	}
 }
+
+func TestGetGroups(t *testing.T) {
+	tests := []struct {
+		name         string
+		secret       *api.Secret
+		expectResult []string
+		expectError  bool
+	}{
+		{
+			name: "not set",
+			secret: &api.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: "test"},
+				Data:       map[string][]byte{},
+			},
+			expectResult: []string{bootstrapapi.BootstrapDefaultGroup},
+		},
+		{
+			name: "set to empty value",
+			secret: &api.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: "test"},
+				Data: map[string][]byte{
+					bootstrapapi.BootstrapTokenGroupsKey: []byte(""),
+				},
+			},
+			expectResult: []string{bootstrapapi.BootstrapDefaultGroup},
+		},
+		{
+			name: "invalid prefix",
+			secret: &api.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: "test"},
+				Data: map[string][]byte{
+					bootstrapapi.BootstrapTokenGroupsKey: []byte("foo"),
+				},
+			},
+			expectError: true,
+		},
+		{
+			name: "valid",
+			secret: &api.Secret{
+				ObjectMeta: metav1.ObjectMeta{Name: "test"},
+				Data: map[string][]byte{
+					bootstrapapi.BootstrapTokenGroupsKey: []byte(
+						bootstrapapi.BootstrapDefaultGroup + "," +
+							bootstrapapi.BootstrapGroupPrefix + "foo" + "," +
```

Comment:
nit: weird formatting here?

I think this should be written as it would be in a real BT; as `system:bootstrappers,system:bootstrappers:foo,system:bootstrappers:bar`
If we ever change this prefix it's very reasonable to have to change this unit test.

Cases I want to see covered here:
 - `system:bootstrappers` only
 - `system:bootstrappers:foo` only
 - `system:bootstrappers:foo,system:bootstrappers:bar`
 - `system:bootstrappers,system:bootstrappers:foo,system:bootstrappers:bar` (what you have here)
 - `system:bootstrappers:` (should not be allowed??)
 - `system:bootstrappers:      ` should not be allowed
 - `system:bootstrappers:<script> alert("scary?!") </script>`
 - `system:bootstrappers:foo:` must not end with `:`

etc etc add here whatever you can think of that is useful